api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/core/authentication/ScriptedPrincipalAttributesProperties.java

```diff
@@ -18,6 +18,11 @@
 
     private static final long serialVersionUID = 4221139939506528713L;
 
+    /**
+     * Script engine name, e.g. groovy, js, python, etc. Required if CAS doesn't map based on extension.
+     */
+    private String engineName;
```

Comment:
It might also be good to turn the engine name into an enum, so that validations can occur if someone were to produce a typo, etc.

Reply:
There are a lot of enginenames and many support multiple names (e.g. js and javascript and ecmascript and rhino are all aliases for same engine). jython and python both work as engine names and people could use .jy or .py as the extensions. 

I could dynamically get a list of engines supported by the current classpath and validate the enginename against the various shortnames for all the engines that are found by: 
```
       ScriptEngineManager manager = new ScriptEngineManager();
        List<ScriptEngineFactory> engines = manager.getEngineFactories();
```
This shows sample of getting all the engine aliases:
http://www.drdobbs.com/jvm/jsr-223-scripting-for-the-java-platform/215801163?pgno=1

I wouldn't want to slow down everyone's startup if that is at all expensive but I could try it out. Regarding typos, they will probably get an error message or warning from person directory when it tries to use an invalid engine name.

Reply:
I guess I could just try to get the engine name they entered to see if it is valid, rather than building a full list. That wouldn't penalize anyone that wasn't using the property.

Reply:
Sounds like a good option to me.